OndaEDFSchemas.jl/src/OndaEDFSchemas.jl

```diff
@@ -36,6 +36,9 @@ export PlanV1, PlanV2, FilePlanV1, FilePlanV2, EDFAnnotationV1
 end
 
 Legolas.accepted_field_type(::PlanV1SchemaVersion, ::Type{String}) = AbstractString
+# we need this because Arrow write can introduce a Missing for the error column
+# (I think because of how missing/nothing sentinels are handled?)
+Legolas.accepted_field_type(::PlanV1SchemaVersion, ::Type{Union{Nothing,String}}) = Union{Nothing,Missing,AbstractString}
```

Comment:
Let's try to reduce the code duplication:
```suggestion
const OndaEDFSchemaVersions = Union{PlanV1SchemaVersion,PlanV2SchemaVersion,FilePlanV1SchemaVersion,FilePlanV1SchemaVersion}
Legolas.accepted_field_type(::OndaEDFSchemaVersions, ::Type{String}) = AbstractString
# we need this because Arrow write can introduce a Missing for the error column
# (I think because of how missing/nothing sentinels are handled?)
Legolas.accepted_field_type(::OndaEDFSchemaVersions, ::Type{Union{Nothing,String}}) = Union{Nothing,Missing,AbstractString}
```